unit-tests/run-unit-tests.py

```diff
@@ -179,11 +207,15 @@ def cat( filename ):
     manifestfile = dir + '/CMakeFiles/TargetDirectories.txt'
 else:
     manifestfile = dir + '/../CMakeFiles/TargetDirectories.txt'
-for manifest_ctx in grep( r'(?<=unit-tests/build/)\S+(?=/CMakeFiles/test-\S+.dir$)', manifestfile ):
 
+for manifest_ctx in grep( r'(?<=unit-tests/build/)\S+(?=/CMakeFiles/test-\S+.dir$)', manifestfile ):
+    
     testdir = manifest_ctx['match'].group(0)                    # "log/internal/test-all"
     testparent = os.path.dirname(testdir)                       # "log/internal"
-    testname = 'test-' + testparent.replace( '/', '-' ) + '-' + os.path.basename(testdir)[5:]   # "test-log-internal-all"
+    if testparent != "":
+        testname = 'test-' + testparent.replace( '/', '-' ) + '-' + os.path.basename(testdir)[5:]    # "test-log-internal-all"
+    else:
+        testname = 'test-' + os.path.basename(testdir)[5:]      # no parent folder so we get "test-all"
```

Comment:
I believe we use `[5:] to take everything past the `test-`... but here we append it to `test-` anyway.... so why not just simply take the whole thing?

Reply:
You are correct, I changed it